extensions/healthie/actions/__tests__/getPatient.ts

```diff
@@ -0,0 +1,318 @@
+            },
+          },
+          onComplete,
+          jest.fn()
+        )
+
+        expect(onComplete).toHaveBeenCalledWith({
+          data_points: {
+            firstName: returnValue.data.user.first_name,
+            lastName: returnValue.data.user.last_name,
+            dob: DATE_FNS_STRING,
+            email: returnValue.data.user.email,
+            gender: returnValue.data.user.gender,
+            phoneNumber: validatedPhone,
+            groupName: returnValue.data.user.user_group.name,
+            primaryProviderId: returnValue.data.user.dietitian_id,
+          },
+          events: undefined,
+        })
+      }
+    )
+
+    test.each([
+      { healthiePhone: '', validatedPhone: undefined },
+      { healthiePhone: undefined, validatedPhone: undefined },
+    ])(
+      '$#. Should log event and return "undefined" for validatedPhone when healthie returns "$healthiePhone"',
+      async ({ healthiePhone, validatedPhone }) => {
+        const returnValue = mockGetSdkReturn.getUser({})
+        mockGetSdkReturn.getUser.mockReturnValueOnce({
+          data: {
+            user: {
+              ...returnValue.data.user,
+              phone_number: healthiePhone as string,
+            },
+          },
+        })
+
+        await getPatient.onActivityCreated(
+          {
+            pathway: {
+              id: 'pathway-id',
+              definition_id: 'pathway-definition-id',
+            },
+            activity: {
+              id: 'activity-id',
+            },
+            patient: { id: 'test-patient' },
+            fields: {
+              patientId: 'patient-1',
+            },
+            settings: {
+              apiKey: 'apiKey',
+              apiUrl: 'test-url',
+            },
+          },
+          onComplete,
+          jest.fn()
+        )
+
+        expect(onComplete).toHaveBeenCalledWith({
+          data_points: {
+            firstName: returnValue.data.user.first_name,
+            lastName: returnValue.data.user.last_name,
+            dob: DATE_FNS_STRING,
+            email: returnValue.data.user.email,
+            gender: returnValue.data.user.gender,
+            phoneNumber: validatedPhone,
+            groupName: returnValue.data.user.user_group.name,
+            primaryProviderId: returnValue.data.user.dietitian_id,
+          },
+          events: [
+            {
+              date: DATE_MOCK.toISOString(),
+              error: {
+                category: 'WRONG_DATA',
+                message:
+                  "Phone number from Healthie not stored because it isn't a valid E.164 phone number",
+              },
+              text: {
+                en: "Phone number from Healthie not stored because it isn't a valid E.164 phone number",
+              },
+            },
+          ],
+        })
+      }
+    )
+  })
+
+  describe('dob validation', () => {
+    test.each([
+      {
+        healthiePhone: '+48 123 456 789',
+        validatedPhone: '+48123456789',
```

Comment:
[Q] The test description doesn't match your test setup as you are passing in both phones & dates in `test.each`. Can this test run by only passing `healthieDate` and `validatedDate` in the `test.each` array instead?

Reply:
The mock returns patient object. Inside action there is a validation for phone and date and accordingly parsed value is assigned or event is sent. I need to have specific values so that action doesn't send event for phone validation error. The only thing I can try is to hardcode that value entirely. I'll try that.